test/splitter.js

```diff
@@ -0,0 +1,160 @@
+const Splitter = artifacts.require("./Splitter.sol");
+
+contract('Splitter', async accounts => {
+
+    let gasCost = async receipt => {
+        const gasUsed = receipt.receipt.gasUsed;
+        const tx = await web3.eth.getTransaction(receipt.tx);
+        const gasPrice = tx.gasPrice;
+
+        return web3.utils.toBN(gasUsed * gasPrice);
+    };
+
+    let checkEventNotEmitted = async eventName => {
+        let result = await truffleAssert.createTransactionResult(splitter, splitter.transactionHash);
+
+        await truffleAssert.eventNotEmitted(
+            result,
+            eventName
+        );
+    }
+
+    const senderAddress = accounts[1];
+    const RecipientOneAddress = accounts[2];
+    const RecipientTwoAddress = accounts[3];
+    let splitter;
+
+    beforeEach("deploy and prepare", async function() {
+        splitter = await Splitter.new();
+    });
+
+    it('Sender sends 5 wei, 4 wei is split equally between recipients and 1 wei sent to the senders balance', async () => {
+
+        // Take snapshot of initial balances
+        let initSenderEthBalance = await web3.eth.getBalance(senderAddress);
+        initSenderEthBalance = web3.utils.toBN(initSenderEthBalance);
+
+        const initSenderContractBalance = await splitter.balances(senderAddress);
+        const initRecipientOneContractBalance = await splitter.balances(RecipientOneAddress);
+        const initRecipientTwoContractBalance = await splitter.balances(RecipientTwoAddress);
+
+        // Perform transactions
+        const receipt = await splitter.splitDeposit(RecipientOneAddress, RecipientTwoAddress, {from: senderAddress, value: 5});
+
+        const cost = await gasCost(receipt);
+        const transfer = web3.utils.toBN(5);
+        const splitAmount = web3.utils.toBN(2);
+        const remainder = web3.utils.toBN(1);
+
+        // Check sender's changed ETH balance
+        const expectedSenderEthBalance = initSenderEthBalance.sub(cost).sub(web3.utils.toBN(transfer));
+        let aliceEthBalance = await web3.eth.getBalance(senderAddress);
+        aliceEthBalance = web3.utils.toBN(aliceEthBalance);
+
+        assert.strictEqual(aliceEthBalance.toString(10), expectedSenderEthBalance.toString(10));
+
+        // Calculate the expected contract balances
+        const expectedSenderContractBalance = initSenderContractBalance.add(remainder);
+        const expectedRecipientOneContractBalance = initRecipientOneContractBalance.add(splitAmount);
+        const expectedRecipientTwoContractBalance = initRecipientTwoContractBalance.add(splitAmount);
+
+        // Get the actual contract balances
+        const aliceContractBalance = await splitter.balances(senderAddress);
+        const bobContractBalance = await splitter.balances(RecipientOneAddress);
+        const carolContractBalance = await splitter.balances(RecipientTwoAddress);
+
+        assert.strictEqual(aliceContractBalance.toString(10), expectedSenderContractBalance.toString(10));
```

Comment:
To me since `initSenderContractBalance` is `0`, then here what we are testing is `, "1");`. The others below are `, "2");` for the same reason.

Reply:
I understand what you mean. I have made this unnecessarily verbose.